### PR TITLE
feat(activity): scaffold redesigned ActivityScreen behind tmcuActivityRedesignEnabled flag

### PR DESCRIPTION
**Pull request by vinnyhoward**
## **Description**

Scaffolds the redesigned Activity screen behind the `tmcuActivityRedesignEnabled` LaunchDarkly flag. When the flag is off, the existing `ActivityView` renders unchanged.

The new screen mirrors the extension's activity-v3 work in [MetaMask/metamask-extension#42837](https://github.com/MetaMask/metamask-extension/pull/42837) and lays the foundation for the unified transaction list once mobile adapters land:

- Context-aware empty state ready to be plugged in with adapters
- Stacked/collapsing header (`HeaderStandardAnimated`) with back button
- Search input (`TextFieldSearch`) — UI only, no functionality yet
- Horizontally-scrollable filter row with network + type chips
  - Network filter reuses `TrendingTokenNetworkBottomSheet`
  - Type filter uses a new `ActivityTypeFilterSheet` with brand-blue active state
- Filter-aware empty states with themed CTAs (Swap / AddFunds / MakePrediction / BrowsePerpsMarkets / TransferToMoney / OpenMetamaskCard); funded vs. unfunded variants gated on `selectAddressHasTokenBalances`
- Theme-aware empty-state SVG illustration (light + dark)
- `ActivityKind` union mirrors the extension's adapter output shape (TODO: replace with import from the core controller lib once available)
- Unit tests for `getActivityEmptyState` + `ActivityTypeFilterSheet`

### ⚠️ Caveats — needs re-validation once dependencies land

- **Empty-state CTA destinations are stubbed.** Each route was chosen by mirroring the closest existing entry point in the codebase (e.g. `useRampNavigation().goToBuy()`, `useMoneyAccountDeposit().initiateDeposit()`, `Routes.CARD.ROOT`, `Routes.PERPS.MARKET_LIST`, `Routes.PREDICT.MARKET_LIST`). These need to be re-verified by the owning teams once the redesign goes live to confirm each CTA lands on the correct flow.
- **Network filter list is a placeholder.** Currently reuses `TRENDING_NETWORKS_LIST` from the Trending screen. This must be replaced with the canonical Activity network list once the adapters define which chains are actually available.
- **No real transactions yet.** Filter state (`networkFilter`, `typeFilter`, `searchQuery`) is wired and ready, but the list itself remains a TODO until adapters land.

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes: [TMCU-861](https://consensyssoftware.atlassian.net/browse/TMCU-861)

## **Manual testing steps**

```gherkin
Feature: Activity screen redesign behind feature flag

  Scenario: Flag off — legacy view renders
    Given the LaunchDarkly flag `tmcuActivityRedesignEnabled` is false
    When the user navigates to the Activity screen
    Then the existing tabbed ActivityView renders unchanged

  Scenario: Flag on — new screen renders
    Given the LaunchDarkly flag `tmcuActivityRedesignEnabled` is true
    When the user navigates to the Activity screen
    Then the redesigned ActivityScreen renders with header, search, and filter chips

  Scenario: Type filter
    Given the redesigned ActivityScreen is open
    When the user taps the "All types" chip and selects "Perps"
    Then the bottom sheet dismisses
    And the chip label updates to "Types: Perps" with brand-blue tint
    And the empty state copy becomes "Your first perps trade could be your best."
    And tapping "Browse markets" navigates to the Perps market list

  Scenario: Network filter
    Given the redesigned ActivityScreen is open
    When the user taps the "All networks" chip and selects "BNB Chain"
    Then the bottom sheet dismisses
    And the chip label updates to "Network: BNB Chain" with brand-blue tint

  Scenario: Funded vs. unfunded empty state
    Given the user's aggregated token balance is zero
    When the redesigned ActivityScreen renders with no filter
    Then the empty state shows "Add funds" copy and CTA
    And when funds are added the copy switches to "Swap tokens"
```

## **Screenshots/Recordings**

https://github.com/user-attachments/assets/9f267a8b-61fa-4f1e-9d35-4df6c98e7035

### **Before**

`~`

### **After**

`~`

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

#### Performance checks (if applicable)

- [x] I've tested on Android
  - Ideally on a mid-range device; emulator is acceptable
- [x] I've tested with a power user scenario
  - Use these [power-user SRPs](https://consensyssoftware.atlassian.net/wiki/spaces/TL1/pages/edit-v2/401401446401?draftShareId=9d77e1e1-4bdc-4be1-9ebb-ccd916988d93) to import wallets with many accounts and tokens
- [x] I've instrumented key operations with Sentry traces for production performance metrics
  - See [`trace()`](/app/util/trace.ts) for usage and [`addToken`](/app/components/Views/AddAsset/components/AddCustomToken/AddCustomToken.tsx#L274) for an example

For performance guidelines and tooling, see the [Performance Guide](https://consensyssoftware.atlassian.net/wiki/spaces/TL1/pages/400085549067/Performance+Guide+for+Engineers).

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

[TMCU-854]: https://consensyssoftware.atlassian.net/browse/TMCU-854?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

> [!NOTE]
> **Medium Risk**
> Swaps the Activity tab UI when the flag is enabled and routes empty-state CTAs into buy, bridge, perps, predict, money, and card flows; legacy behavior is preserved when the flag is off.
> 
> **Overview**
> Adds a **feature-flagged** Activity redesign: when `tmcuActivityRedesignEnabled` is on, `ActivityView` renders new `ActivityScreen` instead of the existing tabbed UI (renamed `LegacyActivityView`).
> 
> The new screen scaffolds **animated header**, **search UI** (state only), **network + type filter chips** with bottom sheets (`ActivityTypeFilterSheet` and a stub `TrendingTokenNetworkBottomSheet`), and a **filter-aware empty state** whose copy/CTAs depend on type filter and `selectAddressHasTokenBalances`. It wires `ActivityTypeFilter` → `ActivityKind` mappings for future unified list filtering; the transaction list itself is still a TODO.
> 
> Also adds `selectIsActivityRedesignEnabled`, E2E test IDs, light/dark empty illustrations, expanded `activity_view` strings, and unit tests for empty-state resolution and the type filter sheet.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit af13f0672272acfcf5e3d0514f73730269287d0f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

[TMCU-861]: https://consensyssoftware.atlassian.net/browse/TMCU-861?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/components/Views/ActivityScreen/ActivityScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+                icon={
+                  <ActivityEmptyIcon
+                    name="activity-empty"
+                    width={72}
+                    height={78}
+                  />
+                }
+                description={strings(emptyState.descriptionKey)}
+                actionButtonText={strings(emptyState.actionLabelKey)}
+                onAction={handleEmptyStateAction}
```
Comment (n3ps) [inline app/components/Views/ActivityScreen/ActivityScreen.tsx]: With this having some logic, how about encapsulating the Empty state into its own component? 

Ex: it can take in the type filter than decide what to render and the action

File: app/components/Views/ActivityScreen/types.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  | 'nftMint'
+  | 'lendingDeposit'
+  | 'lendingWithdrawal'
+  | 'predictionsAddFunds'
+  | 'predictionsWithdrawFunds'
+  | 'predictionClaimWinnings'
+  | 'predictionCashedOut'
+  | 'predictionPlaced'
+  | 'perpsAddFunds'
+  | 'perpsWithdrawFunds'
```
Comment (n3ps) [inline app/components/Views/ActivityScreen/types.ts]: nit: incoming PR in extension to rename this to just `perpsWithdraw`

**Review by n3ps — DISMISSED** [review]

File: app/components/Views/ActivityScreen/types.ts (hunk 2 above)
Comment (vinnyhoward) [reply app/components/Views/ActivityScreen/types.ts]: Will update
